examples/treeple/treeple_tutorial_pvalue_multiview.py

```diff
@@ -0,0 +1,238 @@
+    stratify=True,
+    random_state=1,
+)
+
+# fit the model and obtain the tree posteriors
+_, single_proba = build_hyppo_oob_forest(est, Z, y)
+
+# generate forest posteriors for the two classes
+single_proba = np.nanmean(single_proba, axis=0)
+
+
+# scatter plot the posterior probabilities for class one
+plt.hist(single_proba[:500][:, 1], bins=30, alpha=0.6, color="blue", label="negative")
+plt.hist(single_proba[500:][:, 1], bins=30, alpha=0.6, color="red", label="positive")
+plt.legend()
+plt.show()
+
+#
+# Generate null posteriors with Z and permuted X
+# ----------------------------------------------
+
+
+# shuffle the labels
+X_null = np.copy(X)
+np.random.shuffle(X_null)
+
+# initialize another forest with 100 trees
+est_null = HonestForestClassifier(
+    n_estimators=100,
+    max_samples=1.6,
+    max_features=0.3,
+    bootstrap=True,
+    stratify=True,
+    tree_estimator=MultiViewDecisionTreeClassifier(),
+    random_state=1,
+)
+
+# fit the model and obtain the tree posteriors
+_, null_proba = build_hyppo_oob_forest(est, np.hstack((X_null, Z)), y)
+
+# generate forest posteriors for the two classes
+null_proba = np.nanmean(null_proba, axis=0)
+
+
+# scatter plot the posterior probabilities for class one
+plt.hist(null_proba[:500][:, 1], bins=30, alpha=0.6, color="blue", label="negative")
+plt.hist(null_proba[500:][:, 1], bins=30, alpha=0.6, color="red", label="positive")
+plt.legend()
+plt.show()
+
+
+# %%
+# Find the observed statistic difference
+# --------------------------------------
+def Calculate_MI(y_true, y_pred_proba):
+    # calculate the conditional entropy
+    H_YX = np.mean(entropy(y_pred_proba, base=np.exp(1), axis=1))
+
+    # empirical count of each class (n_classes)
+    _, counts = np.unique(y_true, return_counts=True)
+    # calculate the entropy of labels
+    H_Y = entropy(counts, base=np.exp(1))
+    return H_Y - H_YX
+
+
+joint_mi = Calculate_MI(y, observe_proba)
+mi = Calculate_MI(y, single_proba)
+cmi = joint_mi - mi
+
+joint_mi_null = Calculate_MI(y, null_proba)
+cmi_null = joint_mi_null - mi
+
+observed_diff = cmi - cmi_null
+print("Observed statistic difference =", round(observed_diff, 2))
+
+# %%
+# Permute the trees
+# -----------------
+
+
+PERMUTE = 10000
+mix_diff = []
+
+# Collect all the tree posteriors
+proba = np.vstack((observe_proba, null_proba))
+for i in range(PERMUTE):
+
+    # permute the posteriors
+    np.random.shuffle(proba)
+
+    # calculate the statistic for
+    # the two mixed forest posteriors
+    mi_mix_one = Calculate_MI(y, proba[:100])
+    mi_mix_two = Calculate_MI(y, proba[100:])
+
+    # same difference of joint MI and CMI
+    mix_diff.append(mi_mix_one - mi_mix_two)
+
+
+# %%
+# Calculate the p-value
+# ---------------------
+
+
+pvalue = (1 + (mix_diff >= observed_diff).sum()) / (1 + PERMUTE)
+
+print("p-value is:", round(pvalue, 2))
+if pvalue < 0.05:
+    print("The null hypothesis is rejected.")
+else:
+    print("The null hypothesis is not rejected.")
```

Comment:
Should we remove these LOC? https://github.com/neurodata/scikit-tree/pull/256/files/2c487755d8a0827b7344b2bee6da8eabef1d400e#r1567919332